File: projects/quarkus-stock/src/main/java/com/vjcspy/stockinfo/model/cor/CorMapper.java
package com.vjcspy.stockinfo.model.cor;

public class CorMapper {

    public static CorDto toDTO(CorEntity entity) {
        CorDto dto = new CorDto();
        dto.id = entity.id;
        dto.code = entity.code;
        dto.exchange = entity.exchange;
        dto.name = entity.name;
        dto.refId = entity.refId;
        dto.catId = entity.catId;
        dto.industryName1 = entity.industryName1;
        dto.industryName2 = entity.industryName2;
        dto.industryName3 = entity.industryName3;
        dto.totalShares = entity.totalShares;
        dto.firstTradeDate = entity.firstTradeDate;
        return dto;
    }

    public static CorEntity toEntity(CorDto dto) {
        CorEntity entity = new CorEntity();
        entity.id = dto.id;
        entity.code = dto.code;
        entity.exchange = dto.exchange;
        entity.name = dto.name;
        entity.refId = dto.refId;
        entity.catId = dto.catId;
        entity.industryName1 = dto.industryName1;
        entity.industryName2 = dto.industryName2;
        entity.industryName3 = dto.industryName3;
        entity.totalShares = dto.totalShares;
        entity.firstTradeDate = dto.firstTradeDate;
        return entity;
    }
}

File: projects/stock/src/main/kotlin/com/vjcspy/spring/packages/stockinfo/mapper/CorMapper.kt
// (mr.vjcspy@gmail.com) 2024
package com.vjcspy.spring.packages.stockinfo.mapper

import com.vjcspy.spring.packages.stockinfo.dto.CorEntityDto
import com.vjcspy.spring.packages.stockinfo.entity.CorEntity
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.factory.Mappers

@Mapper
interface CorMapper {
    companion object {
        val INSTANCE: CorMapper = Mappers.getMapper(CorMapper::class.java)
    }

    // Mapping từ Entity sang DTO
    @Mapping(source = "id", target = "id")
    @Mapping(source = "refId", target = "refId")
    @Mapping(source = "catId", target = "catId")
    @Mapping(source = "code", target = "code")
    @Mapping(source = "exchange", target = "exchange")
    @Mapping(source = "industryName1", target = "industryName1")
    @Mapping(source = "industryName2", target = "industryName2")
    @Mapping(source = "industryName3", target = "industryName3")
    @Mapping(source = "totalShares", target = "totalShares")
    @Mapping(source = "name", target = "name")
    @Mapping(source = "firstTradeDate", target = "firstTradeDate")
    fun toDto(entity: CorEntity): CorEntityDto

    // Mapping từ DTO sang Entity
    @Mapping(source = "id", target = "id")
    @Mapping(source = "refId", target = "refId")
    @Mapping(source = "catId", target = "catId")
    @Mapping(source = "code", target = "code")
    @Mapping(source = "exchange", target = "exchange")
    @Mapping(source = "industryName1", target = "industryName1")
    @Mapping(source = "industryName2", target = "industryName2")
    @Mapping(source = "industryName3", target = "industryName3")
    @Mapping(source = "totalShares", target = "totalShares")
    @Mapping(source = "name", target = "name")
    @Mapping(source = "firstTradeDate", target = "firstTradeDate")
    fun toEntity(dto: CorEntityDto): CorEntity
}
